### ISSUE TITLE
Error on run mix phoenix.server

### ISSUE THREAD

**Issue by lmorais** [phoenixframework/phoenix#1003]
mix phoenix.server raising ecto error compile

```
mix phoenix.server
==> ranch (compile)
Compiled src/ranch_transport.erl
Compiled src/ranch_app.erl
Compiled src/ranch_acceptors_sup.erl
Compiled src/ranch_acceptor.erl
Compiled src/ranch_protocol.erl
Compiled src/ranch_listener_sup.erl
Compiled src/ranch_sup.erl
Compiled src/ranch.erl
Compiled src/ranch_conns_sup.erl
Compiled src/ranch_server.erl
Compiled src/ranch_tcp.erl
Compiled src/ranch_ssl.erl
==> poolboy (compile)
Compiled src/poolboy_worker.erl
Compiled src/poolboy_sup.erl
Compiled src/poolboy.erl
==> decimal
Compiled lib/decimal.ex
Generated decimal app
==> poison
Compiled lib/poison.ex
Compiled lib/poison/parser.ex
Compiled lib/poison/decoder.ex
Compiled lib/poison/encoder.ex
Generated poison app
==> cowlib (compile)
Compiled src/cow_http_hd.erl
Compiled src/cow_spdy.erl
Compiled src/cow_date.erl
Compiled src/cow_cookie.erl
Compiled src/cow_http.erl
Compiled src/cow_qs.erl
Compiled src/cow_http_te.erl
Compiled src/cow_multipart.erl
Compiled src/cow_mimetypes.erl
==> cowboy (compile)
Compiled src/cowboy_middleware.erl
Compiled src/cowboy_sub_protocol.erl
Compiled src/cowboy_sup.erl
Compiled src/cowboy_loop_handler.erl
Compiled src/cowboy.erl
Compiled src/cowboy_static.erl
Compiled src/cowboy_bstr.erl
Compiled src/cowboy_handler.erl
Compiled src/cowboy_spdy.erl
Compiled src/cowboy_clock.erl
Compiled src/cowboy_protocol.erl
Compiled src/cowboy_app.erl
Compiled src/cowboy_router.erl
Compiled src/cowboy_websocket_handler.erl
Compiled src/cowboy_http_handler.erl
Compiled src/cowboy_rest.erl
Compiled src/cowboy_http.erl
Compiled src/cowboy_websocket.erl
Compiled src/cowboy_req.erl
==> plug
Compiled lib/plug.ex
Compiled lib/plug/adapters/cowboy.ex
Compiled lib/plug/conn/adapter.ex
Compiled lib/plug/builder.ex
Compiled lib/plug/conn/cookies.ex
Compiled lib/plug/conn/unfetched.ex
Compiled lib/plug/conn/status.ex
Compiled lib/plug/conn/query.ex
Compiled lib/plug/crypto.ex
Compiled lib/plug/conn/wrapper_error.ex
Compiled lib/plug/crypto/key_generator.ex
Compiled lib/plug/conn/utils.ex
Compiled lib/plug/crypto/message_encryptor.ex
Compiled lib/plug/crypto/message_verifier.ex
Compiled lib/plug/error_handler.ex
Compiled lib/plug/head.ex
Compiled lib/plug/logger.ex
Compiled lib/plug/adapters/translator.ex
Compiled lib/plug/adapters/test/conn.ex
Compiled lib/plug/adapters/cowboy/handler.ex
Compiled lib/plug/exceptions.ex
Compiled lib/plug/method_override.ex
Compiled lib/plug/router.ex
Compiled lib/plug/parsers.ex
Compiled lib/plug/parsers/urlencoded.ex
Compiled lib/plug/conn.ex
Compiled lib/plug/parsers/json.ex
Compiled lib/plug/router/utils.ex
Compiled lib/plug/session.ex
Compiled lib/plug/csrf_protection.ex
Compiled lib/plug/session/store.ex
Compiled lib/plug/session/ets.ex
Compiled lib/plug/session/cookie.ex
Compiled lib/plug/supervisor.ex
Compiled lib/plug/static.ex
Compiled lib/plug/test.ex
Compiled lib/plug/debugger.ex
Compiled lib/plug/parsers/multipart.ex
Compiled lib/plug/adapters/cowboy/conn.ex
Compiled lib/plug/upload.ex
Compiled lib/plug/mime.ex
Generated plug app
==> phoenix
Compiled lib/mix/tasks/compile.phoenix.ex
Compiled lib/mix/phoenix.ex
Compiled lib/mix/tasks/phoenix.routes.ex
Compiled lib/mix/tasks/phoenix.gen.resource.ex
Compiled lib/mix/tasks/phoenix.server.ex
Compiled lib/mix/tasks/phoenix.new.ex
Compiled lib/phoenix.ex
Compiled lib/phoenix/code_reloader/change_detector.ex
Compiled lib/phoenix/code_reloader/proxy.ex
Compiled lib/phoenix/code_reloader/server.ex
Compiled lib/phoenix/code_reloader.ex
Compiled lib/phoenix/config.ex
Compiled lib/phoenix/controller/pipeline.ex
Compiled lib/phoenix/endpoint.ex
Compiled lib/phoenix/endpoint/adapter.ex
Compiled lib/phoenix/endpoint/cowboy_handler.ex
Compiled lib/phoenix/endpoint/cowboy_websocket.ex
Compiled lib/phoenix/endpoint/watcher.ex
Compiled lib/phoenix/endpoint/server.ex
Compiled lib/phoenix/html/controller.ex
Compiled lib/phoenix/exceptions.ex
Compiled lib/phoenix/html.ex
Compiled lib/phoenix/html/engine.ex
Compiled lib/phoenix/controller.ex
Compiled lib/phoenix/naming.ex
Compiled lib/phoenix/controller/logger.ex
Compiled lib/phoenix/html/tag.ex
Compiled lib/phoenix/html/link.ex
Compiled lib/phoenix/pubsub.ex
Compiled lib/phoenix/pubsub/local.ex
Compiled lib/phoenix/html/form_data.ex
Compiled lib/phoenix/html/form.ex
Compiled lib/phoenix/pubsub/pg2.ex
Compiled lib/phoenix/html/safe.ex
Compiled lib/phoenix/pubsub/pg2_server.ex
Compiled lib/phoenix/pubsub/redis.ex
Compiled lib/phoenix/pubsub/redis_conn.ex
Compiled lib/phoenix/param.ex
Compiled lib/phoenix/pubsub/redis_server.ex
Compiled lib/phoenix/router/resource.ex
Compiled lib/phoenix/router/route.ex
Compiled lib/phoenix/router/helpers.ex
Compiled lib/phoenix/router/scope.ex
Compiled lib/phoenix/socket.ex
Compiled lib/phoenix/endpoint/error_handler.ex
Compiled lib/phoenix/router/console_formatter.ex
Compiled lib/phoenix/supervisor.ex
Compiled lib/phoenix/router.ex
Compiled lib/phoenix/socket/message.ex
Compiled lib/phoenix/channel.ex
Compiled lib/phoenix/channel/control_channel.ex
Compiled lib/phoenix/template/engine.ex
Compiled lib/phoenix/channel/transport.ex
Compiled lib/phoenix/template/eex_engine.ex
Compiled lib/phoenix/template/exs_engine.ex
Compiled lib/phoenix/template.ex
Compiled lib/phoenix/transports/serializer.ex
Compiled lib/phoenix/transports/json_serializer.ex
warning: Plug.Builder.compile/1 and compile/2 is deprecated, please use compile/3 instead
    lib/phoenix/controller/pipeline.ex:121: Phoenix.Controller.Pipeline."MACRO-__before_compile__"/2
    (elixir) src/elixir_dispatch.erl:187: :elixir_dispatch.expand_macro_fun/6
    (elixir) src/elixir_dispatch.erl:171: :elixir_dispatch.expand_require/5
    (elixir) src/elixir_dispatch.erl:97: :elixir_dispatch.dispatch_require/6
warning: Plug.Builder.compile/1 and compile/2 is deprecated, please use compile/3 instead
    lib/phoenix/controller/pipeline.ex:121: Phoenix.Controller.Pipeline."MACRO-__before_compile__"/2
    (elixir) src/elixir_dispatch.erl:187: :elixir_dispatch.expand_macro_fun/6
    (elixir) src/elixir_dispatch.erl:171: :elixir_dispatch.expand_require/5
    (elixir) src/elixir_dispatch.erl:97: :elixir_dispatch.dispatch_require/6
Compiled lib/phoenix/transports/websocket.ex
Compiled lib/phoenix/transports/long_poller/server.ex
Compiled lib/phoenix/transports/long_poller.ex
Compiled lib/phoenix/view.ex
Generated phoenix app
==> postgrex
Compiled lib/postgrex/binary_utils.ex
Compiled lib/postgrex/error.ex
Compiled lib/postgrex/extension.ex
Compiled lib/postgrex/builtins.ex
Compiled lib/postgrex/result.ex
Compiled lib/postgrex/error_code.ex
Compiled lib/postgrex/type_info.ex
Compiled lib/postgrex/extensions/text.ex
Compiled lib/postgrex/extensions/json.ex
Compiled lib/postgrex/utils.ex
Compiled lib/postgrex/types.ex
Compiled lib/postgrex/messages.ex
Compiled lib/postgrex/extensions/binary.ex
Compiled lib/postgrex/connection.ex
Compiled lib/postgrex/protocol.ex
Generated postgrex app
==> ecto
Compiled lib/ecto/adapter/storage.ex
Compiled lib/ecto/adapter/migration.ex
Compiled lib/ecto.ex
Compiled lib/ecto/adapter/transaction.ex
Compiled lib/ecto/adapter.ex
Compiled lib/ecto/adapters/connection.ex
Compiled lib/ecto/adapters/mysql/connection.ex
Compiled lib/ecto/adapters/pool.ex
Compiled lib/ecto/adapters/poolboy/worker.ex
Compiled lib/ecto/adapters/postgres/datetime.ex
Compiled lib/ecto/adapters/poolboy.ex
Compiled lib/ecto/adapters/sql/query.ex
Compiled lib/ecto/migration/runner.ex
Compiled lib/ecto/migration.ex
Compiled lib/ecto/model.ex
Compiled lib/ecto/migrator.ex
Compiled lib/ecto/model/callbacks.ex
Compiled lib/ecto/query/builder/distinct.ex
Compiled lib/ecto/query/builder/filter.ex
Compiled lib/ecto/query/builder/from.ex
Compiled lib/ecto/exceptions.ex
Compiled lib/ecto/log_entry.ex
Compiled lib/ecto/adapters/sql.ex
Compiled lib/ecto/adapters/sql/sandbox.ex
Compiled lib/ecto/query.ex
Compiled lib/ecto/query/builder.ex
Compiled lib/ecto/adapters/postgres.ex
Compiled lib/ecto/adapters/mysql.ex
Compiled lib/ecto/query/builder/group_by.ex
Compiled lib/ecto/query/builder/join.ex
Compiled lib/ecto/adapters/postgres/connection.ex
Compiled lib/ecto/query/builder/lock.ex
Compiled lib/ecto/query/builder/limit_offset.ex
Compiled lib/ecto/query/builder/select.ex
Compiled lib/ecto/query/builder/order_by.ex
Compiled lib/ecto/query/builder/preload.ex
Compiled lib/ecto/query/builder/update.ex
Compiled lib/ecto/query/inspect.ex
Compiled lib/ecto/repo/assoc.ex
Compiled lib/ecto/repo.ex
Compiled lib/ecto/queryable.ex
Compiled lib/ecto/repo/config.ex
Compiled lib/ecto/query/planner.ex
Compiled lib/ecto/repo/model.ex
Compiled lib/ecto/storage.ex
Compiled lib/ecto/repo/preloader.ex
Compiled lib/mix/ecto.ex
Compiled lib/mix/tasks/ecto.gen.migration.ex
Compiled lib/mix/tasks/ecto.gen.repo.ex
Compiled lib/mix/tasks/ecto.drop.ex
Compiled lib/mix/tasks/ecto.create.ex
Compiled lib/ecto/type.ex
Compiled lib/mix/tasks/ecto.rollback.ex
Compiled lib/ecto/uuid.ex
Compiled lib/mix/tasks/ecto.migrate.ex
Compiled lib/ecto/schema.ex
Compiled lib/ecto/repo/queryable.ex
Compiled lib/ecto/changeset.ex
Compiled lib/ecto/model/optimistic_lock.ex
Compiled lib/ecto/model/timestamps.ex
Compiled lib/ecto/association.ex
Compiled lib/ecto/migration/schema_migration.ex
Compiled lib/ecto/datetime.ex
Generated ecto app
==> phoenix_ecto
Compiled lib/phoenix_ecto.ex
Compiled lib/phoenix_ecto/plug.ex

== Compilation error on file lib/phoenix_ecto/html.ex ==
** (CompileError) lib/phoenix_ecto/html.ex:6: unknown key :errors for struct Phoenix.HTML.Form
    (elixir) src/elixir_map.erl:182: :elixir_map."-assert_struct_keys/5-lc$^0/1-0-"/5
    (elixir) src/elixir_map.erl:59: :elixir_map.translate_struct/4
    (elixir) src/elixir_clauses.erl:36: :elixir_clauses.clause/7
    (elixir) src/elixir_def.erl:178: :elixir_def.translate_clause/7
    (elixir) src/elixir_def.erl:167: :elixir_def.translate_definition/8

could not compile dependency phoenix_ecto, mix compile failed. You can recompile this dependency with `mix deps.compile phoenix_ecto` or update it with `mix deps.update phoenix_ecto`

```

**Comment by chrismccord** [comment]
Can you provide your elixir, phoenix, and ecto versions? 

    $ elixir -v
    $ mix deps

**Comment by lmorais** [comment]
The error occurred in version 0.10.0
I'm fix it installing 0.14.0 version with command bellow:

```
mix archive.install https://github.com/phoenixframework/phoenix/releases/download/v0.14.0/phoenix_new-0.14.0.ez

```

Thanks! ;)
